streaming_providers/debrid_client.py

```
@@ -1,121 +1,194 @@
+        await self.initialize_headers()
+        return self
 
-    def __del__(self):
-        if self.token:
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        if self.token and not self.is_private_token:
             try:
-                self.disable_access_token()
+                await self.disable_access_token()
             except ProviderException:
                 pass
```

Ruff

Use `contextlib.suppress(ProviderException)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(ProviderException)`

(SIM105)